### PR TITLE
webview: power by proto instead of JSON marshaling

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: internal/hud/webview/convert.go [hunk 1]
```diff
@@ -414,12 +407,14 @@ func protoPopulateResourceInfoView(mt *store.ManifestTarget, r *proto_webview.Re
 
 func runtimeStatus(res ResourceInfoView) RuntimeStatus {
 	_, isLocal := res.(LocalResourceInfo)
-	if isLocal {
+	_, isLocalPtr := res.(*LocalResourceInfo)
```
Comment (nicks) [inline internal/hud/webview/convert.go]: this seems like it's working around a bug that should be fixed upstream. i don't see a legitimate reason to accept both types here

File: internal/hud/webview/view.go [hunk 2]
```diff
@@ -118,17 +123,27 @@ func ToWebViewBuildRecord(br model.BuildRecord) BuildRecord {
 	}
 }
 
+func timeToProto(t time.Time) *timestamp.Timestamp {
+	ts, err := ptypes.TimestampProto(t)
+	if err != nil {
+		return nil
```
Comment (nicks) [inline internal/hud/webview/view.go]: this seems dangerous. i would be more comfortable with this if it was a `panic()`, tho it really should return an error

Comment (jazzdan) [reply internal/hud/webview/view.go]: @nicks the proto know show to handle `nil` as the empty value for a timestamp, so I think this is OK.

https://github.com/golang/protobuf/blob/master/ptypes/timestamp.go#L95

Comment (nicks) [reply internal/hud/webview/view.go]: oh, my concern isn't that we're returning nil, but that we're quietly dropping validation errors

Comment (jazzdan) [reply internal/hud/webview/view.go]: And in fact empty values is what the web UI expects here.

Comment (jazzdan) [reply internal/hud/webview/view.go]: Ah, okay. I convert all of these things to return error then.

**Comment by jazzdan** [comment]
cc @landism converting the Facets to protobuf

**Pull request merged**